src/host/main.cpp

```diff
@@ -475,9 +500,28 @@ int main(int argc, char** argv)
     logger::config::initialize_with_json_console();
   }
 
-  if (max_open_sessions_hard == 0)
+  // Fill in derived default values
+  if (!(*public_rpc_address_option))
+  {
+    public_rpc_address = rpc_address;
+  }
+
+  if (!(*mosh_option))
   {
-    max_open_sessions_hard = max_open_sessions + hard_session_cap_diff;
+    max_open_sessions_hard =
+      max_open_sessions + cli::ParsedRpcInterface::default_mosh_diff;
+  }
+
+  // If --rpc-address etc were specified, they populate a single object at the
+  // start of the rpc_interfaces list
+  if (*rpc_address_option)
```

Comment:
We still have the original CLI args `--rpc-address`, `--max-open-sessions` etc. They allow you to specify a single interface, that could have been a `--rpc-interface` option. I think its worth keeping this complication for now, assuming that when we do #2612 and have a tool to convert CLI to JSON, we will flatten into a single form there.